sonar-ws/src/test/java/org/sonarqube/ws/client/component/SearchProjectsRequestTest.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonarqube.ws.client.component;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class SearchProjectsRequestTest {
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  SearchProjectsRequest.Builder underTest = SearchProjectsRequest.builder();
+
+  @Test
+  public void default_page_values() {
+    SearchProjectsRequest result = underTest.build();
+
+    assertThat(result.getPage()).isEqualTo(1);
+    assertThat(result.getPageSize()).isEqualTo(100);
+  }
+
+  @Test
+  public void handle_paging_limit_values() {
+    SearchProjectsRequest result = underTest.setPageSize(500).build();
+
+    assertThat(result.getPage()).isEqualTo(1);
+    assertThat(result.getPageSize()).isEqualTo(500);
+  }
+
+  @Test
+  public void fail_if_page_size_greater_than_500() {
+    expectedException.expect(IllegalArgumentException.class);
```

Comment:
without testing the message, we don't know if IAE was caused by the page size check or anything else